# Module 10.1: Linear Regression Basics

In [4]:
# Import libraries
import math
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from pandas.core.common import random_state
from sklearn.linear_model import LinearRegression

### LOS 10.a: Describe a simple linear regression model, how the least squares criterion is used to estimate regression coefficients, and the interpretation of these coefficients.

The purpose of simple linear regression is to explain the variation in a dependent variable in terms of the variation in a single independent variable. Here, the term variation is interpreted as the degree to which a variable differs from its mean value. Don't confuse variation with variance—they are related, but they are not the same.


$\text{variation in Y} = {\Large\sum_{1}^{n}} (Y_i - \bar{Y})^2$

* The **dependent variable** is the variable whose variation is explained by the independent variable. We are interested in answering the question, "What explains fluctuations in the dependent variable?" The dependent variable is also referred to as the terms explained variable, endogenous variable, or predicted variable.
</br>

* The **independent variable** is the variable used to explain the variation of the dependent variable. The independent variable is also referred to as the terms explanatory variable, exogenous variable, or predicting variable.
</br>

**Example: Dependent vs. independent variables**

Suppose you want to predict stock returns with GDP growth. Which variable is the independent variable?

&emsp;**Answer:**

Because GDP is going to be used as a predictor of stock returns, stock returns are being *explained* by GDP. Hence, stock returns are the dependent (explained) variable, and GDP is the independent (explanatory) variable.


#### Simple Linear Regression Model

The following linear regression model is used to describe the relationship between two variables, $X$ and $Y$:
<br></br>
$\Large{Y_i = b_0 + b_1X_i + \epsilon_i ,... i = 1, ..., n}$

&emsp; 

<U>where:</U>

$Y_i$ = ith observation of the dependent variable, $Y$ 

$X_i$ = ith observation of the independent variable, $X$
 
$b_0$ = regression intercept term
 
$b_1$ = regression slope coefficient
 
$\epsilon_i$ = **residual** for the $i_{th}$ observation (also referred to as the disturbance term or error term);

Based on this regression model, the regression process estimates an equation for a line through a scatter plot of the data that "best" explains the observed values for $Y$ in terms of the observed values for $X$.


#### Simple Linear Regression Model

The linear equation, often called the line of best fit or regression line, takes the following form:
<br></br>

$\Large\hat{Y}_{i} = \hat{b}_{0} + \hat{b}_{1}X_i i=1,2,3...,n$

&emsp; 

<U>where:</U>

$\hat{Y}_{i}$ = estimated value of $Y_i$ given $X_i$

$\hat{b}_{0}$ = estimated intercept term.

$\hat{b}_{1}$ = estimated slope coefficient.

 
<br>
The hat "^" above a variable or parameter indicates a predicted value.
</br>


Thus, the regression line is the line that minimizes the **SSE**. This explains why simple linear regression is frequently referred to as ordinary least squares **(OLS) regression**, and the values determined by the estimated regression equation, $\hat{Y}_i$, are called least squares estimates.

<br>
The estimated slope coefficient $\hat{b}_{1}$ for the regression line describes the change in $Y$ for a one-unit change in $X$. It can be positive, negative, or zero, depending on the relationship between the regression variables. The slope term is calculated as follows:
</br>
&emsp; 

$\Large\hat{b}_{1} = \frac{CovXY}{\sigma^2_X}$

The intercept term $\hat{b}_{0}$ is the line's intersection with the $Y$-axis at $X = 0$. It can be positive, negative, or zero. A property of the least squares method is that the intercept term may be expressed as follows:

&emsp; 
$\large\hat{b}_{0}=\bar{Y}−\hat{b}_{1}\bar{X}$

where:

Y = mean of Y

X = mean of X

The intercept equation highlights the fact that the regression line passes through a point with coordinates equal to the mean of the independent and dependent variables (i.e., the point X, Y).

<hr>

**Example: Computing the slope coefficient and intercept term**

Compute the slope coefficient and intercept term using the following information:

<table>
<thead>
  <tr>
    <th></th>
    <th></th>
    <th></th>
    <th></th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Cov(S&amp;P 500, ABC</td>
    <td>0.000336</td>
    <td>Mean return, S&amp;P 500</td>
    <td>−2.70%</td>
  </tr>
  <tr>
    <td>Var(S&amp;P 500)</td>
    <td>0.000522</td>
    <td>Mean return, ABC</td>
    <td>−4.05%</td>
  </tr>
</tbody>
</table>

**Answer:**

The slope coefficient is calculated as $\hat{b}_{1} = \frac{0.000336}{0.000522} = 0.64$.

The intercept term is calculated as follows:

$\hat{b}_{0}=\overline{ABC}−\hat{b}_{1}\overline{SP500}=−4.05\% −0.64 (−2.70\%) = −2.3\%$
<br>
</br>
The estimated regression line that minimizes the SSE in our ABC stock return example is shown in  Estimated Regression Equation for ABC vs. S&P 500 Excess Returns.

<br>
This regression line has an intercept of $–2.3\%$ and a slope of $0.64$. The model predicts that if the S&P 500 excess return is $–7.8\%$ (May 20X4 value), then the ABC excess return would be $–2.3\% + (0.64)(–7.8\%) = –7.3\%$. The residual (error) for the May 20X4 ABC prediction is $8.4\%$—the difference between the actual ABC excess return of $1.1\%$ and the predicted return of $–7.3\%$.
</br>


In [5]:
covAB= 0.000336
ABC_Var = 0.000522
SP500_Mu = -2.70
ABC_Mu = -4.05
## Define the Slope intercept
bhat_1 = covAB / ABC_Var
bhat_0 = ABC_Mu - (bhat_1*SP500_Mu)

#Excess returns
excessSP = bhat_0 + bhat_1 * -7.8

#Actual ABC returns
actABC = 1.1
sse = actABC - excessSP 

In [6]:
print("Slope Coeffiecient b^1   =  ", round(bhat_1, 2))
print("Intercept b^0            = ", round(bhat_0, 1))
print("Excess returns of SP500  = ", round(excessSP, 1))
print("Actual returns of ABC    =  ", round(actABC, 1))
print("SSE Error ABC prediction =  ", round(sse, 1))

Slope Coeffiecient b^1   =   0.64
Intercept b^0            =  -2.3
Excess returns of SP500  =  -7.3
Actual returns of ABC    =   1.1
SSE Error ABC prediction =   8.4


<hr>


<img src="https://github.com/PachaTech/CFA-Level-1/blob/main/Module%2010/pics/10_1%20graph.jpeg?raw=true">

<br>



A Simple linear regression explains the variation in a dependent variable $Y$ in terms of the variation in a single independent variable $X$

#### Assumptions of a Simple Linear Regression

1. The relationship between $X$ independent and $Y$ dependent does (must) exist. 
2. Error terms are normally distrubuted. Their $\mu=0$
3. The variance $\sigma$ of the error term is constant *(Homoskedastic)*.
4. Error terms are independently distributed and uncorrelated with each other, *(serial or autocorrelation)*.  
5. Error terms are not random.

<hr></hr>

#### Results can be an issue for Standard Error terms or to Hypothesis testing.
* HOMOSKEDACITY - refers to the case where all prediction errors all have the same constant variance.  $\sigma = c$
<br>

* HETEROSKEDACITY - refers to the variance of the error terms $\epsilon$ not being constant.    $\sigma \neq c$
<br>

* Conditional HETEROSKEDACITY -  where the variance of the error terms is related to the independent variable.  
*i.e. if the independent variable is getting bigger and bigger or the variance is increasing and getting bigger.  Maybe, the independent variable is getting smaller and the variance is getting smaller too.*
<br>

#### NOTES
* The model **does not** assume that the dependent variable is uncorrelated with the residuals. 
* The model **does assume** that the independent variable is uncorrelated with the residuals


### Module 10.1: Linear Regression Basics Quiz
<hr>
1. What is the most appropriate interpretation of a slope coefficient estimate equal to 10.0?

* **Answer:** For every 1-unit change in the independent variable, the model predicts that the dependent variable will change by 10 units.
 

* **Explanation:** The slope coefficient is best interpreted as the predicted change in the dependent variable for a 1-unit change in the independent variable; if the slope coefficient estimate is 10.0 and the independent variable changes by 1 unit, the dependent variable is expected to change by 10 units. The intercept term is best interpreted as the value of the dependent variable when the independent variable is equal to zero. (Module 10.1, LOS 10.a)

<hr>

2. Which of the following is least likely a necessary assumption of simple linear regression analysis?


* **Answer:** The dependent variable is uncorrelated with the residuals.
    
    
* **Explanation:** The model does not assume that the dependent variable is uncorrelated with the residuals. It does assume that the independent variable is uncorrelated with the residuals. (Module 10.1, LOS 10.b)

## Module 10.2: Analysis of Variance (ANOVA) and Goodness of Fit

<hr>

### LOS 10.c: Calculate and interpret measures of fit and formulate and evaluate tests of fit and of regression coefficients in a simple linear regression.


### LOS 10.d: Describe the use of analysis of variance (ANOVA) in regression analysis, interpret ANOVA results, and calculate and interpret the standard error of estimate in a simple linear regression.

<hr>

**Analysis of variance (ANOVA)** is a statistical procedure for analyzing the total variability of the dependent variable. Let's define some terms before we move on to ANOVA tables:

* The **total sum of squares (SST)** measures the total variation in the dependent variable. SST is equal to the sum of the squared differences between the actual $Y$-values and the mean of $Y$:

$\qquad\large\text{SST}=\displaystyle\sum_{i=1}^n(Y_i−\overline{Y})^2$

* The **sum of squares regression (SSR)** measures the variation in the dependent variable that is explained by the independent variable. SSR is the sum of the squared distances between the predicted $Y$-values and the mean of $Y$:

$\qquad\large\text{SSR}=\displaystyle\sum_{i=1}^n(\hat{Y}−\overline{Y})^2$

* The **mean square regression (MSR)** is the SSR divided by the number of independent variables. A simple linear regression has only one independent variable, so in this case, MSR = SSR.

#### Professor's Note

Multiple regression (i.e., with more than one independent variable) is addressed in the Level II CFA curriculum.

* The **sum of squared errors (SSE)** measures the unexplained variation in the dependent variable. It's also known as the sum of squared residuals or the residual sum of squares. SSE is the sum of the squared vertical distances between the actual $Y$-values and the predicted $Y$-values on the regression line:
<br>

$\qquad\large\text{SSR}=\displaystyle\sum_{i=1}^n(Y_i−\hat{Y})^2$

</br>

* The **mean squared error (MSE)** is the SSE divided by the degrees of freedom, which is $n – 1$ minus the number of independent variables. A simple linear regression has only one independent variable, so in this case, degrees of freedom are $n – 2$.

<br>

You probably will not be surprised to learn the following:
<br>

$\qquad\text{total variation = explained variation + unexplained variation}$

<br>
or:

<br>
$\qquad\text{SST = SSR + SSE}$

<br>

**Components of Total Variation** illustrates how the total variation in the dependent variable (SST) is composed of SSR and SSE.


<img src="https://github.com/PachaTech/CFA-Level-1/blob/main/Module%2010/pics/10_2%20chart.jpeg?raw=true">

<br>

<hr>

The output of the ANOVA procedure is an ANOVA table, which is a summary of the variation in the dependent variable. ANOVA tables are included in the regression output of many statistical software packages. You can think of the ANOVA table as the source of the data for the computation of many of the regression concepts discussed in this reading. A generic ANOVA table for a simple linear regression (one independent variable) is presented in  **ANOVA Table for a Simple Linear Regression**.

<img src="https://github.com/PachaTech/CFA-Level-1/blob/main/Module%2010/pics/10_2.jpeg?raw=true">
<br>

#### Standard Error of Estimate (SEE)

The SEE for a regression is the standard deviation of its residuals. The lower the SEE, the better the model fit:

$\qquad\Large\text{SEE} = \sqrt{MSE}$


$\qquad\Large\text{SEE} = \sqrt{\frac{SSE}{n-2}}$

#### Coefficient of Determination (R2)

The coefficient of determination (R2) is defined as the percentage of the total variation in the dependent variable explained by the independent variable. For example, an R2 of 0.63 indicates that the variation of the independent variable explains 63% of the variation in the dependent variable:

$\qquad\Large{R^2 = \frac{\text{SSR}}{\text{SST}}}$

#### Professor's Note

For simple linear regression (i.e., with one independent variable), the coefficient of determination, $R^2$, may be computed by simply squaring the correlation coefficient, $r$. In other words, $R^2 = r^2$ for a regression with one independent variable.

**Example:** Using the ANOVA table

Given the following ANOVA table based on 36 observations, calculate the $R^2$ and the standard error of estimate **(SEE)**.

<hr>



<center><b>Completed ANOVA table for ABC regression<\center><\b>

<table>
<thead>
  <tr>
    <th>Source of Variation</th>
    <th>Degrees of Freedom</th>
    <th>Sum of Squares</th>
    <th>Mean Sum of Squares</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Regression (explained)</td>
    <td>1</td>
    <td>0.0076</td>
    <td>0.0076</td>
  </tr>
  <tr>
    <td>Error (unexplained)</td>
    <td>34</td>
    <td>0.0406</td>
    <td>0.0012</td>
  </tr>
  <tr>
    <td>Total</td>
    <td>35</td>
    <td>0.0482</td>
    <td></td>
  </tr>
</tbody>
</table>

In [7]:
## Input the variables
SSR = 0.0076
SST = 0.0482
MSE = 0.0012
# Solve for MSE
SSE = 0.0012

## Solve for coefficient of determination (R^2)
R2 = (SSR / SST)

## Calculate SEE
SEE = math.sqrt(MSE)

In [8]:
## Print variables
print("Explained Variation (SSR)          =  ", round(SSR, 6))
print("Total Variation     (SST)          =  ", round(SST, 7))
print("coefficient of determination (MSE) =  ", round(R2, 3))
print("coefficient of determination (R^2) =  %", round(R2*100,1))
print("Standard Error of Estimate (SEE)   =  ", round(SEE, 3))

Explained Variation (SSR)          =   0.0076
Total Variation     (SST)          =   0.0482
coefficient of determination (MSE) =   0.158
coefficient of determination (R^2) =  % 15.8
Standard Error of Estimate (SEE)   =   0.035


**Answer**

$\quad R^2=\frac{\text{explained variation (SSR)}}{\text{total variation (SST)}} = \frac{0.0076}{0.0482}=0.158$ or 15.8% 

<br>

$\qquad\text{SEE}=\sqrt{MSE}=\sqrt{0.0012}=0.035$


#### The F-Statistic

An *F*-test assesses how well a set of independent variables, as a group, explains the variation in the dependent variable.

The *F*-statistic is calculated as follows:

<br>
$\quad\large\text{F}=\frac{\text{MSR}}{\text{MSE}}=\frac{\text{SSR}\div K}{\text{SSE}\div(n-k-1)}$

where:

$\text{MSR} =$ mean regression sum of squares

$\text{MSE} =$ mean squared error

**Important**: This is always a one-tailed test

<br>
For simple linear regression, there is only one independent variable, so the $F$-test is equivalent to a $t$-test of the statistical significance of the slope coefficient:

$\qquad H_0: b_1 = 0$   versus   $H_a: b_1 \neq 0$

To determine whether $b_1$ is statistically significant using the $F$-test, the calculated $F$-statistic is compared with the critical $F$-value, $F_c$, at the appropriate level of significance. The degrees of freedom for the numerator and denominator with one independent variable are as follows:

$\qquad df_{\text{numerator}} = k = 1$

$\qquad df_{\text{denominator}} = n − k − 1 = n − 2$

where:

$n =$ number of observations

The decision rule for the $F$-test is to reject $H_0$ if $F > F_c$.

<br>

Rejecting the null hypothesis that the value of the slope coefficient equals zero at a stated level of significance indicates that the independent variable and the dependent variable have a significant linear relationship.
<hr>

In [9]:
## Define variables
k = 1
n = -2

# Solve the F-test
MSR = SSR / k
F = MSR / MSE 
# MSE = SEE / (n-k-1) 


## print(k)
## print(n)
## print(MSR)
## print(MSE)
print("F-Test Decision Rule (F)   =  ", round(F, 2))

F-Test Decision Rule (F)   =   6.33


**Example**: Calculating and interpreting the $F$-statistic

Use the ANOVA table from the previous example to calculate and interpret the F-statistic. Test the null hypothesis at the 5% significance level that the slope coefficient is equal to 0.

**Answer**:

$\qquad\Large F=\frac{\text{MSR}}{\text{MSE}}= \frac{0.0076}{0.0012}=6.33$

$\text{df}_\text{numerator} = k = 1$

$\text{df}_\text{denominator} = n − k − 1 = 36 − 1 − 1 = 34$


The null and alternative hypotheses are $h_0:b_1 = 0$ versus $h_a:b_1 \neq 0$. 
The critical $F$-value for $1$ and $34$ degrees of freedom at a $5\%$ significance level is approximately $4.1$. 
(Remember, it's a one-tailed test, so we use the $5\%$ $F$-table.) 
Therefore, we can reject the null hypothesis and conclude that the slope coefficient is significantly different than zero.

**Hypothesis Test of a Regression Coefficient**

A t-test may also be used to test the hypothesis that the true slope coefficient, $b_1$, is equal to a hypothesized value. Letting $\hat{b}_1$ be the point estimate for $b_1$, the appropriate test statistic with $n − 2$ degrees of freedom is:

$\qquad  tb_1= \Large\frac{\hat{b}_1 - {b}_1}{S\hat{b}_1} $

The decision rule for tests of significance for regression coefficients is:

Reject $H_0$ if $t > + $ $t_\text{critical}$ or $t < -$ $t_\text{critical}$

Rejection of the null supports the alternative hypothesis that the slope coefficient is different from the hypothesized value of $b_1$. To test whether an independent variable explains the variation in the dependent variable (i.e., it is statistically significant), the null hypothesis is that the true slope is zero $(b_1 = 0)$. The appropriate test structure for the null and alternative hypotheses is:

$H_0: b_1 = 0$ versus  $H_a:b_1 \neq 0 $

**Example: Hypothesis test for significance of regression coefficients**

The estimated slope coefficient from the ABC example is $0.64$ with a standard error equal to $0.26$. Assuming that the sample has $36$ observations, determine if the estimated slope coefficient is significantly different than zero at a $5\%$ level of significance.

$\color{red}{\text{NOTE: Use the T-table in the back of the book for the observations and variables}}$

In [10]:
## Define Variables
bhat = 0.64
b1 = 0
s_b1 = 0.26


## Solve for t
t = (bhat - b1) / s_b1

## Print Output
print("t = ", round(t, 2))

t =  2.46


**Answer**:

The calculated test statistic is:

$\Large t=\Large\frac{\hat{b}_1-b_1}{S_{\hat{b}_1}}= \frac{0.64-0}{0.26}= 2.46$

The critical two-tailed $t$-values are  $\pm2.03$ (from the $t$-table with df = 36 − 2 = 34). Because $t > \text{t}_\text{critical}$ (i.e., $2.46 > 2.03$), we reject the null hypothesis and conclude that the slope is different from zero.

Note that the $t$-test for a simple linear regression is equivalent to a $t$-test for the correlation coefficient between $x$ and $y$:

$\qquad\Large t_2 = \Large\frac{r\sqrt{n-2}}{\sqrt{1-r2}}$

## Module 10.2: Analysis of Variance (ANOVA) and Goodness of Fit Quiz
<HR>
    
1. Consider the following statement: "In a simple linear regression, the appropriate degrees of freedom for the critical t-value used to calculate a confidence interval around both a parameter estimate and a predicted Y-value is the same as the number of observations minus two". This statement is:
    
    * **Answer:** justified
    * **Explanation:** In simple linear regression, the appropriate degrees of freedom for both confidence intervals is the number of observations in the sample (n) minus two. (Module 10.2, LOS 10.c)

<br>    
    
2. What is the appropriate alternative hypothesis to test the statistical significance of the intercept term in the following regression?
$\qquad Y = a_1 + a_2(X) + \epsilon$
    
    * **Answer:** $H_A: a_1 \neq 0$.
    * **Explanation:** In this regression, a1 is the intercept term. To test the statistical significance means to test the null hypothesis that a1 is equal to zero, versus the alternative that a1 is not equal to zero. (Module 10.2, LOS 10.c)

<br>    
    
3. The variation in the dependent variable explained by the independent variable is measured by the:
    
    * **Answer:** regression sum of squares.
    * **Explanation:** The regression sum of squares measures the amount of variation in the dependent variable explained by the independent variable (i.e., the explained variation). The sum of squared errors measures the variation in the dependent variable not explained by the independent variable. The mean squared error is equal to the sum of squared errors divided by its degrees of freedom. (Module 10.2, LOS 10.d)
    
    

## Module 10.3: Predicted Values and Functional Forms of Regression
<hr>

### LOS 10.e: Calculate and interpret the predicted value for the dependent variable, and a prediction interval for it, given an estimated linear regression model and a value for the independent variable.

**Predicted values** are values of the dependent variable based on the estimated regression coefficients and a prediction about the value of the independent variable. They are the values that are predicted by the regression equation, given an estimate of the independent variable.

For a simple regression, this is the predicted (or forecast) value of $Y$:

$\hat{Y}=\hat{b}_0+\hat{b}_1 X_p$

where:

$\hat{Y} =$ predicted value of the dependent variable

$X_p =$ forecasted value of the independent variable

<hr>

**Example:** Predicting the dependent variable

Given the ABC regression equation as follows:

$\widehat{ABC} = −2.3\% + (0.64) (\widehat{SP500}) $

Calculate the predicted value of ABC excess returns if forecast S&P 500 excess returns are 10%.

**Answer:**

The predicted value for ABC excess returns is determined as follows:

$\widehat{ABC} = −2.3\% + (0.64)(10\%) = 4.1\%$



#### Confidence Intervals for Predicted Values

This is the equation for the confidence interval for a predicted value of Y:

$\hat{Y}\pm (t_c \times s_f)\Rightarrow [\hat{Y}(t_c \times s_f) < Y < \hat{Y} + (t_c \times s_f)]$

where:

$t_c$ = two-tailed critical t-value at the desired level of significance with df = n − 2

$s_f$ = standard error of the forecast




In [20]:
## Define Forecast Values for equities
abc_hat = -0.023 + (bhat6 * 0.1) 
print("The forecast for ABC excess returns is: ",(round(abc_hat*100, 8)),"%")## Calculate formula 

The forecast for ABC excess returns is:  4.1 %


<mark>
$\color{blue}{\text{The intercept (-2.3%) & Slope coefficient (0.64) are sample values}}$
    
 * **two sources of error**.
     1. the missing error term
     2. slope & intercept are sample, not population values.
        * Confidence Intervals
        * Standard error of the estimate <b>SEE</b>
        * The <b>SEE</b> is the standard deviation of the error term


The challenge with computing a <u>confidence interval</u> for a predicted value is calculating $s_f$. 
On the Level I exam, it's highly unlikely that you will have to calculate the standard error of the forecast (it <b>will probably be provided</b> if you need to compute a confidence interval for the dependent variable). However, if you do need to calculate sf, it can be done with the following formula for the variance of the forecast:
    
##### Confidence Error    

$ \hat{Y}\pm(t_c \times s_f)   \longleftarrow   s^2_f=\text{SEE}^2 \Biggr[1+\large\frac{1}{n}+(X−\overline{X})^2\large\frac{(n−1)}{s^2_x}\Biggr]  $

    
where:

$\text{SEE}^2 =$ variance of the residuals = the square of the standard error of estimate

$s^2_x =$ variance of the independent variable

$X =$ value of the independent variable for which the forecast was made

<hr>

**Example:** Confidence interval for a predicted value

Calculate a $95\%$ prediction interval on the predicted value of ABC excess returns from the previous example. Assume the standard error of the forecast is $3.67$, and the forecast value of S&P 500 excess returns is $10\%$.

**Answer:**

This is the predicted value for ABC excess returns:

$\widehat{ABC}=−2.3\%+(0.64)(10\%)=4.1\%$

The 5% two-tailed critical t-value with 34 degrees of freedom is 2.03. This is the prediction interval at the 95% confidence level:


In [28]:
#ABC HAT Confidence intervals Returns
abc_hat2 = (abc_hat) + 0.0203 * 0.0367
print("The forecast for ABC Confidence Interval returns is: ",(round(abc_hat2*100, 10)),"%")

The forecast for ABC Confidence Interval returns is:  4.174501 %


$\widehat{ABC}\pm(t_c \times s_f)\Rightarrow[4.1\% \pm, (2.03×3.67\%)]=4.1\% \pm 7.5\%$

Or, $–3.4\%$ to $11.6\%$.

We can interpret this range to mean that, given a forecast value for S&P 500 excess returns of $10\%$, we can be $95\%$ confident that the ABC excess returns will be between $–3.4\%$ and $11.6\%$.


### Types of Models
In such a situation, transforming one or both of the variables can produce a linear relationship. The appropriate transformation depends on the relationship between the two variables. One often-used transformation is to take the natural log of one or both of the variables. Here are some examples:

* **Log-lin model**. This is if the dependent variable is logarithmic, while the independent variable is linear.


* **Lin-log model**. This is if the dependent variable is linear, while the independent variable is logarithmic.


* **Log-log model**. Both the dependent variable and the independent variable are logarithmic.


Selecting the correct functional form involves determining the nature of the variables and evaluating the goodness-of-fit measures (e.g., R2, SEE, F-stat).


#### Log-Lin Model

Taking the natural logarithm of the dependent variable, our model now becomes this:

$\ln Y_i = b_0 + b_1(X)_i + \epsilon_i$

In this model, the slope coefficient is interpreted as the relative change in dependent variable for an absolute change in the independent variable.  Log-Lin Model, EPS Data shows the results after taking the natural log of EPS, and fitting that data using a log-lin model.

* **Log-lin model**. This is if the dependent variable is logarithmic, while the independent variable is linear.

#### Log-Log Model

Taking the natural logarithm of both variables, our model now becomes this:

$\ln Y_i = b_0 + b_1\ln(X)_i + \epsilon_i$

In this model, the slope coefficient is interpreted as the relative change in dependent variable for a relative change in the independent variable.

* **Log-log model**. Both the dependent variable and the independent variable are logarithmic.



#### Lin-Log Model

Taking the natural logarithm of the independent variable, our model now becomes this:

$Y_i = b_0 + b_1\ln(X)_i + \epsilon_i$

In this model, the slope coefficient is interpreted as the absolute change in dependent variable for a relative change in the independent variable.


* **Lin-log model**. This is if the dependent variable is linear, while the independent variable is logarithmic.




#### 